src/missing.jl

```diff
@@ -0,0 +1,47 @@
+"""
+Exclude any missing indices from being included the wrappped distance metric.
+"""
+struct SkipMissing{D<:PreMetric} <: PreMetric
```

Comment:
I'm not sure if there's a benefit to have specific subtypes here? All the metrics I tested seemed to work.